remove boost_cropped spkg, add boost to prereqs

dimpase opened the pull request:
boost is available on all the supported platforms, and maintaining the spkg thus makes no sense any more. So e.g #34748 can be closed

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

dimpase committed:
move boost to _prereqs, purge boost_cropped spkg

antonio-rojas commented on build/pkgs/_prereq/distros/arch.txt:
```diff
@@ -20,3 +20,5 @@ bzip2
 patch
 pkgconf
+zlib
+libboost-devel
```
This package doesn't exist on Arch. To correct one is `boost`

dimpase replied:
OK, thanks, will fix.

dimpase replied:
ok, fixed

dimpase committed:
it's boost on arch

dimpase committed:
ask for boost version 1.71 or newer

dimpase committed:
libboost-dev to docker

dimpase committed:
zeromq opensuse.txt

trying to make sure all needed stuff gets in

dimpase committed:
trying boost-base in opensuse

dimpase committed:
tweak boost version for opensuse

dimpase committed:
leave it boost-base for OpenSuse

dimpase committed:
push version back down to 1.66

dimpase committed:
more in  SPKG.rst on C compilers and Boost

dimpase committed:
sphinx doesn't like loose '*'

dimpase committed:
Update arch.txt

vbraun committed:
Define a SAGE_CONFIGURE_BOOST environment variable in the build shell

This makes the top-level configure --with-boost=... available for
package configurations. It is then passed down into brial's configure.

vbraun committed:
Pass the boost path from configure --with-boost down to meson

vbraun committed:
Clean artifact, and improve configure help

dimpase committed:
do not check for an nonexistent boost library

dimpase committed:
boost_root in .ini cannot be assigned an empty string

dimpase committed:
re-add lost boost-related rectification of SPKG.rst
